### PR TITLE
[stable] Added linux-stable watcher

### PR DESCRIPTION
**Pull request by mbivolan**

### CODE REVIEW

File: linux_pipeline/Jenkinsfile_stable_kernel_trigger [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env groovy
+
+properties ([
+    pipelineTriggers([cron('H 12 * * *')])
+])
+
+env.WORK_DIR = "/mnt/tmp/watcher-kernel-stable"
+env.KERNEL_TREE = "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git"
+env.VERSIONS = "v4.16-rc|v4.17-rc"
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_stable_kernel_trigger]: shouldn't we check for v4.16.y and v4.17.y?

File: linux_pipeline/Jenkinsfile_stable_kernel_trigger [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+env.DISTRO = "Ubuntu_16.04.3;CentOS_7.4"
+env.LISA_XML = "kernel_pipeline_bvt.xml"
+env.STAGES = "build_artifacts, publish_temp_artifacts, boot_test, publish_artifacts, validation_functional, validation_functional_hyperv, publish_azure_vhd, validation_functional_azure, validation_perf_azure"
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_stable_kernel_trigger]: without big flavor?

File: scripts/package_building/check_latest_tags.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    fi
+}
+
+function get_latest_commits {
+    repo_name="$1"
+    versions="$2"
+    results_path="$3"
+    
+    pushd "./${repo_name}"
+    full_tags="$(git tag -l --format='%(refname) %(taggerdate)' | grep "$(date +"%b %d")" | grep $(date +"%Y"))"
```
Comment (ader1990) [inline scripts/package_building/check_latest_tags.sh]: can we substract one day?

it is safer to check all the tags from previous day, as the day has been completed

File: scripts/package_building/check_latest_tags.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        esac
+    done
+    
+    if [[ ! $WORK_DIR || ! $KERNEL_TREE || ! $RESULTS_PATH ]];then
+        exit 1
+    fi
+    if [[ ! -d $WORK_DIR ]];then
+        mkdir -p $WORK_DIR
+    fi
+    cat /dev/null > $RESULTS_PATH
```
Comment (ader1990) [inline scripts/package_building/check_latest_tags.sh]: use if -e and remove and recreate it if it does not exist

File: scripts/package_building/check_latest_tags.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    versions="$2"
+    results_path="$3"
+    
+    pushd "./${repo_name}"
+    full_tags="$(git tag -l --format='%(refname) %(taggerdate)' | grep "$(date +"%b %d")" | grep $(date +"%Y"))"
+    if [[ $versions ]];then
+        full_tags="$(echo "$full_tags" | grep -E "$versions")"
+    fi
+    tags="$(echo "$full_tags" | sed 's/ .*//' )"
+    commits=""
```
Comment (papagalu) [inline scripts/package_building/check_latest_tags.sh]: if this wants to be an array, declare it with "declare -a ARRAYNAME"

File: scripts/package_building/check_latest_tags.sh [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                shift 2;;
+            --results)
+                RESULTS_PATH="$(readlink -ef $2)"
+                shift 2;;
+            *) break ;;
+        esac
+    done
+    
+    if [[ ! $WORK_DIR || ! $KERNEL_TREE || ! $RESULTS_PATH ]];then
+        exit 1
```
Comment (papagalu) [inline scripts/package_building/check_latest_tags.sh]: maybe print some error message, so we know what went wrong

File: linux_pipeline/Jenkinsfile_stable_kernel_trigger [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+properties ([
+    pipelineTriggers([cron('H 1 * * *')])
+])
+
+env.WORK_DIR = "/mnt/tmp/watcher-kernel-stable"
+env.KERNEL_TREE = "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git"
+env.DISTRO = "Ubuntu_16.04.3;CentOS_7.4"
+env.AZURE_BOOT_LARGE_FLAVOR = "Standard_E64_v3"
+env.LISA_XML = "kernel_pipeline_bvt.xml"
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_stable_kernel_trigger]: let's run full validation - FVT

**Pull request merged**